### PR TITLE
removed code relative to cnecs in series with a pst

### PR DESCRIPTION
**Pull request by phiedw**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Removing a feature for which the specifications were misunderstood

**What is the current behavior?**

We do not include in the functional part of the objective function cnecs that are linked with a pst in the "do-not-optimize-cnec-secured-by-its-pst" parameter, when taps would be sufficient to solve constraints on that cnec.
This isn't exactly what the TSOs wanted.

**What is the new behavior (if this is a feature change)?**
The feature is removed.

**Does this PR introduce a breaking change or deprecate an API?**
- [x] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [x] The *Breaking Change* or *Deprecated* label has been added
- [x] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

You should remove the parameter "do-not-optimize-cnec-secured-by-its-pst".
If you were using that feature, please contact us so we can understand why you were using this feature, and what exactly the need for it is on your end, so we can try find a better solution for you.

**Other information**:

### CODE REVIEW

**Comment by phiedw** [comment]
I'm not sure what relevant test I could add.

File: ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/commons/parameters/UnoptimizedCnecParameters.java [hunk 1]
```diff
@@ -7,117 +7,31 @@
...
-            return new UnoptimizedCnecParameters(
-                operatorsNotSharingCras,
-                null);
-        } else if (!parameters.getDoNotOptimizeCnecsSecuredByTheirPst().isEmpty()) {
-            return new UnoptimizedCnecParameters(
-                    null,
-                    getDoNotOptimizeCnecsSecuredByTheirPstFromIds(parameters.getDoNotOptimizeCnecsSecuredByTheirPst(), crac));
+    public static UnoptimizedCnecParameters build(NotOptimizedCnecsParameters parameters, Set<String> operatorsNotSharingCras) {
+        if (parameters.getDoNotOptimizeCurativeCnecsForTsosWithoutCras()) {
+            return new UnoptimizedCnecParameters(operatorsNotSharingCras);
```
[the remark this answers is not in the document]
Comment (bqth29) [reply ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/commons/parameters/UnoptimizedCnecParameters.java:32]: Also can you one-line this method?

File: ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/UnoptimizedCnecFiller.java [hunk 2]
```diff
@@ -34,59 +28,40 @@
...
  * unless they are worse than their pre-perimeter margins.
- * * selectedRule = PST_LIMITATION : some cnecs parametrized as in series with a pst will not be taken into account in the objective
- * function, as long as a there are enough setpoints on the pst left to absorb the margin deficit
  *
  * @author Peter Mitri {@literal <peter.mitri at rte-france.com>}
  * @author Godelaine de Montmorillon {@literal <godelaine.demontmorillon at rte-france.com>}
  */
 public class UnoptimizedCnecFiller implements ProblemFiller {
     public static final String VARIABLE_NOT_CREATED = "%s variable has not yet been created for Cnec %s (side %s)";
     public static final String OPTIMIZE_CNEC_BINARY = "Optimize cnec binary";
```
[the remark this answers is not in the document]
Comment (phiedw) [reply ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/UnoptimizedCnecFiller.java:40]: No. I can't even seem to find where it was used before.

File: ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/UnoptimizedCnecFiller.java [hunk 3]
```diff
@@ -34,59 +28,40 @@
...
     private final double highestThresholdValue;
-    private final Map<FlowCnec, RangeAction<?>> flowCnecRangeActionMap;
-    private final RangeActionsOptimizationParameters rangeActionParameters;
-    private UnoptimizedCnecFillerRule selectedRule;
 
     public UnoptimizedCnecFiller(OptimizationPerimeter optimizationContext,
                                  Set<FlowCnec> flowCnecs,
                                  FlowResult prePerimeterFlowResult,
                                  UnoptimizedCnecParameters unoptimizedCnecParameters,
                                  RangeActionsOptimizationParameters rangeActionParameters) {
```
[the remark this answers is not in the document]
Comment (phiedw) [reply ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/UnoptimizedCnecFiller.java:50]: indeed

File: ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/UnoptimizedCnecFiller.java [hunk 4]
```diff
@@ -34,59 +28,40 @@
...
         this.operatorsNotToOptimize = unoptimizedCnecParameters.getOperatorsNotToOptimize();
         this.highestThresholdValue = RaoUtil.getLargestCnecThreshold(flowCnecs, MEGAWATT);
-        this.flowCnecRangeActionMap = unoptimizedCnecParameters.getDoNotOptimizeCnecsSecuredByTheirPst();
-        this.rangeActionParameters = rangeActionParameters;
     }
 
     public enum UnoptimizedCnecFillerRule {
         MARGIN_DECREASE,
         PST_LIMITATION
     }
```
[the remark this answers is not in the document]
Comment (phiedw) [reply ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/UnoptimizedCnecFiller.java:61]: I guess not

File: ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/commons/parameters/UnoptimizedCnecParameters.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (phiedw) [reply ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/commons/parameters/UnoptimizedCnecParameters.java:32]: All the other parameters have builders so it allows for slightly more homogeneity

**Review by bqth29 — DISMISSED** [review]

**Review by pet-mit — APPROVED** [review]

**Pull request merged**